.github/workflows/build_packages.yml

```diff
@@ -7,7 +7,21 @@
 name: Build packages
 
 on:
+  # Trigger from another workflow (typically to build dev packages and then test them)
+  workflow_call:
+    inputs:
+      build_type:
+        description: The type of build version to produce ("stable", "rc", or "dev")
+        type: string
+        default: "dev"
```

Comment:
> This is refactors the `build_packages.yml` workflow so it can be used via `workflow_call` as part of a "pkgci" setup, as an alternative to creating a new `pkgci_build_packages.yml` workflow as originally proposed in #589. This lets us reuse the same workflow for building stable, nightly, and dev packages, all across the same matrix of Python versions and operating systems. Package builds take about 2 minutes (wall time) across the full matrix, so we might as well build them all, instead of artificially constraining ourselves to a subset like only Linux on Python 3.11.

Sample workflow run with that setup:

https://github.com/ScottTodd/shark-ai/actions/runs/12246364049

![image](https://github.com/user-attachments/assets/3d550e17-2e5b-4085-b695-e37ca0dd2cf2)